src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animals) {
+    if (!animals.hidden && animals instanceof Herbivore) {
+      animals.health -= 50;
```

Comment:
plural `animals` misleads when reading your code
you're not working with plural (array, list etc.) - i know `animal` is reserved already, but you can come up with another relevant singular to maintain code readability - say, `victim` / `creature` etc.

Reply:
i will thank you